### PR TITLE
feat(linear-progress): Implement Linear Progress indicators

### LINKED ISSUE #29: Implement Linear Progress Indicator Component
Spec reference: https://www.google.com/design/spec/components/progress-activity.html#progress-activity-types-of-indicators

See also: https://github.com/PolymerElements/paper-linear-progress/pull/1

> Copied from https://github.com/google/material-design-lite/issues/4489

### PR DESCRIPTION
**Pull request by lynnjepsen**
Resolves #29: Implement Linear Progress Indicator Component

### CODE REVIEW

File: packages/mdc-linear-progress/mdc-linear-progress.scss [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  &--accent .mdc-linear-progress__bar-inner {
+    @include mdc-theme-prop(background-color, accent);
+  }
+
+  &__buffering-dots {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    background-image: url(data:image/svg+xml;charset=utf8,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20x%3D%270px%27%20y%3D%270px%27%20enable-background%3D%27new%200%200%205%202%27%20xml%3Aspace%3D%27preserve%27%20viewBox%3D%270%200%205%202%27%20preserveAspectRatio%3D%27none%20slice%27%3E%3Ccircle%20cx%3D%271%27%20cy%3D%271%27%20r%3D%271%27%20fill%3D%27%23e6e6e6%27%2F%3E%3C%2Fsvg%3E);
```
Comment (amsheehan) [inline packages/mdc-linear-progress/mdc-linear-progress.scss]: NICE! The only thing is apparently you can squeak out a bit more performance by not encoding single quotes or spaces 👍 

https://codepen.io/tigt/post/optimizing-svgs-in-data-uris

File: packages/mdc-linear-progress/mdc-linear-progress.scss [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    background-size: 10px 4px;
+    animation: buffering 250ms infinite linear;
+  }
+
+  &__buffer {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    transform-origin: top left;
+    transition: transform 250ms ease;
```
Comment (amsheehan) [inline packages/mdc-linear-progress/mdc-linear-progress.scss]: `mdc-animation`

File: demos/linear-progress.html [hunk 3]
```diff
@@ -0,0 +1,186 @@
+<!DOCTYPE html>
+<!--
+  Copyright 2016 Google Inc. All rights reserved.
```
Comment (amsheehan) [inline demos/linear-progress.html]: 2017

File: packages/mdc-linear-progress/constants.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline packages/mdc-linear-progress/constants.js]: 2017

File: packages/mdc-linear-progress/foundation.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline packages/mdc-linear-progress/foundation.js]: 2017

File: packages/mdc-linear-progress/index.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline packages/mdc-linear-progress/index.js]: 2017

File: packages/mdc-linear-progress/mdc-linear-progress.scss [hunk 7]
```diff
@@ -0,0 +1,132 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline packages/mdc-linear-progress/mdc-linear-progress.scss]: 2017

File: test/unit/mdc-linear-progress/foundation.test.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline test/unit/mdc-linear-progress/foundation.test.js]: 2017

File: test/unit/mdc-linear-progress/mdc-linear-progress.test.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (amsheehan) [inline test/unit/mdc-linear-progress/mdc-linear-progress.test.js]: 2017

File: packages/mdc-linear-progress/_keyframes.scss [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+  0% {
+    transform: translateX(0);
+  }
+
+  20% {
+    animation-timing-function: cubic-bezier(.5, 0, .701732, .495819);
+    transform: translateX(0);
+  }
+
+  59.15% {
```
Comment (amsheehan) [inline packages/mdc-linear-progress/_keyframes.scss]: Just looking for clarification, not necessarily that its wrong

Comment (lynnjepsen) [inline packages/mdc-linear-progress/_keyframes.scss]: I just copied all the animation from here
https://component-spec.googleplex.com/progress_and_activity/linear_progress_indicator

**Comment by lynnjepsen** [comment]
Addressed comments. PTAL

File: packages/mdc-linear-progress/index.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  getDefaultFoundation() {
+    return new MDCLinearProgressFoundation({
+      addClass: (className) => this.root_.classList.add(className),
+      getPrimaryBar: () => this.root_.querySelector(MDCLinearProgressFoundation.strings.PRIMARY_BAR_SELECTOR),
+      getBuffer: () => this.root_.querySelector(MDCLinearProgressFoundation.strings.BUFFER_SELECTOR),
+      hasClass: (className) => this.root_.classList.contains(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      setTransform: (el, value) => {
```
Comment (amsheehan) [inline packages/mdc-linear-progress/index.js]: Consider extending `mdc-animation` and extract this out to make the adapter conform more to our recommended [best practices](https://github.com/material-components/material-components-web/blob/master/docs/authoring-components.md#design-adapter-interfaces-to-be-simple-and-intuitive). That said, there's nothing mysterious about what this adapter method is doing, so up to you. :)

File: packages/mdc-linear-progress/mdc-linear-progress.scss [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+  &--accent .mdc-linear-progress__bar-inner {
+    @include mdc-theme-prop(background-color, accent);
+  }
+
+  &__buffering-dots {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    // SVG is optimized for daya URI (https://codepen.io/tigt/post/optimizing-svgs-in-data-uris)
```
Comment (amsheehan) [inline packages/mdc-linear-progress/mdc-linear-progress.scss]: nit: misspelled 'data'

File: packages/mdc-linear-progress/foundation.js [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    this.adapter_.removeClass(cssClasses.CLOSED_CLASS);
+  }
+
+  close() {
+    this.adapter_.addClass(cssClasses.CLOSED_CLASS);
+  }
+
+  setScale(el, scaleValue) {
+    const value = 'scaleX(' + scaleValue + ')';
+    for (let i=0; i<transformStyleProperties.length; i++) {
```
Comment (amsheehan) [inline packages/mdc-linear-progress/foundation.js]: ...forEach?

**Pull request merged**